Sources/SwiftProtobuf/Message+StreamDecodingIterator.swift

```diff
@@ -0,0 +1,158 @@
+// Sources/SwiftProtobuf/Message+StreamDecodingIterator.swift - Iterator over binary delimited input streams
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extensions to `Message` to provide an iterator for decoding binary delimited streams.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+fileprivate let defaultBufferLength = 32768
+
+extension Message {
+  
+  /**
+   Creates an iterator for a binary-delimited protobuf input stream. Errors can be caught by providing a
+   delegate. The returned iterator uses a read-ahead buffer, therefore no assumptions about the position of
+   the stream should be made after any messages have been read.
+   */
```

Comment:
Can we use `///` for doc comments please and document all the parameters of the methods. This follows what we use in other places in this repository.